clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramSaslServer.java

```diff
@@ -143,7 +143,7 @@ public byte[] evaluateResponse(byte[] response) throws SaslException, SaslAuthen
                     } catch (SaslException | AuthenticationException e) {
                         throw e;
                     } catch (Throwable e) {
-                        throw new SaslException("Authentication failed: Credentials could not be obtained", e);
+                        throw new SaslException("Authentication failed: Credentials could not be obtained" + (username == null ? "" : " for user " + username), e);
```

Comment:
From the point the below, we're unsure if `username` is already set or not, so only print it when username has value.